Build failure: keepassxc on Darwin (`error: use of undeclared identifier 'NSAppearanceNameDarkAqua'`)

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `keepassxc` on macOS Intel (tested on 14.1 Sonoma).

### Build log

```
/tmp/nix-build-keepassxc-2.7.6.drv-0/source/src/gui/osutils/macutils/AppKitImpl.mm:84:24: error: property 'effectiveAppearance' not found on object of type '__kindof NSApplication *'
                [NSApp.effectiveAppearance performAsCurrentDrawingAppearance:emitBlock];
                       ^
/tmp/nix-build-keepassxc-2.7.6.drv-0/source/src/gui/osutils/macutils/AppKitImpl.mm:142:19: error: property 'effectiveAppearance' not found on object of type '__kindof NSApplication *'
    return [NSApp.effectiveAppearance.name isEqualToString:NSAppearanceNameDarkAqua];
                  ^
/tmp/nix-build-keepassxc-2.7.6.drv-0/source/src/gui/osutils/macutils/AppKitImpl.mm:142:60: error: use of undeclared identifier 'NSAppearanceNameDarkAqua'; did you mean 'NSAppearanceNameAqua'?
    return [NSApp.effectiveAppearance.name isEqualToString:NSAppearanceNameDarkAqua];
                                                           ^~~~~~~~~~~~~~~~~~~~~~~~
                                                           NSAppearanceNameAqua
/nix/store/1qa2w27s4a47dwgfgvrym2cnaz6xbidq-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSAppearance.h:59:32: note: 'NSAppearanceNameAqua' declared here
APPKIT_EXTERN NSString * const NSAppearanceNameAqua NS_AVAILABLE_MAC(10_9);
                               ^
3 errors generated.
make[2]: *** [src/CMakeFiles/keepassx_core.dir/build.make:2596: src/CMakeFiles/keepassx_core.dir/gui/osutils/macutils/AppKitImpl.mm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:630: src/CMakeFiles/keepassx_core.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
error: builder for '/nix/store/yl0lb85dx52zqfi20c6q6q12snyl48fk-keepassxc-2.7.6.drv' failed with exit code 2;
       last 10 log lines:
       >                                                            ^~~~~~~~~~~~~~~~~~~~~~~~
       >                                                            NSAppearanceNameAqua
       > /nix/store/1qa2w27s4a47dwgfgvrym2cnaz6xbidq-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSAppearance.h:59:32: note: 'NSAppearanceNameAqua' declared here
       > APPKIT_EXTERN NSString * const NSAppearanceNameAqua NS_AVAILABLE_MAC(10_9);
       >                                ^
       > 3 errors generated.
       > make[2]: *** [src/CMakeFiles/keepassx_core.dir/build.make:2596: src/CMakeFiles/keepassx_core.dir/gui/osutils/macutils/AppKitImpl.mm.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:630: src/CMakeFiles/keepassx_core.dir/all] Error 2
       > make: *** [Makefile:166: all] Error 2
```

### Additional context

There seems to be some kind of incompatibility with `AppKit`.

Reverting the update from 2.7.5 to 2.7.6 (392263f17cb85af969a1da8dd79d98ac60a0406c) does not solve the issue, the same error appears.

According to https://developer.apple.com/documentation/appkit/nsappearancenamedarkaqua?language=objc, this option should be available since macOS 10.14.

`keepassxc` is already running on SDK 11:

```nix
  keepassxc = libsForQt5.callPackage ../applications/misc/keepassxc {
    inherit (darwin.apple_sdk_11_0.frameworks) LocalAuthentication;
    stdenv = if stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
  };
```

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@jonafato @rapenne-s @BlankParticle 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-darwin"`
 - host os: `Darwin 23.1.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.5`
 - channels(elian): `"darwin"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/applications/misc/keepassxc/default.nix
+++ b/pkgs/applications/misc/keepassxc/default.nix
@@ -88,7 +88,14 @@ stdenv.mkDerivation rec {
     runHook postCheck
   '';
 
-  nativeBuildInputs = [ asciidoctor cmake wrapGAppsHook wrapQtAppsHook qttools pkg-config ];
+  nativeBuildInputs = [
+    asciidoctor
+    cmake
+    wrapQtAppsHook
+    qttools
+    pkg-config
+  ]
+  ++ lib.optional (!stdenv.isDarwin) wrapGAppsHook;
 
   dontWrapGApps = true;
   preFixup = ''

